packages/core/src/http-client/http-client-types.ts

```diff
@@ -75,7 +75,7 @@ export interface HttpReponse extends KnownHttpResponseFields {
   [otherKey: string]: any;
 }
 
-export interface HttpRequestAndResponse{
-  request: HttpRequest;
-  response: HttpResponse;
+export interface DefaultHttpResponse extends HttpResponse {
+  request: any;
```

Comment:
Why not: `HttpRequest`?
```suggestion
  request: HttpRequest;
```

Reply:
Because the request does not really match `HttpRequest`.
The url (`response.request.url`) and the request body (`response.request.data`) cannot be found from the return object.